runtime/interpreter/statictype.go

```diff
@@ -513,14 +513,17 @@ var _ Authorization = EntitlementSetAuthorization{}
 
 func NewEntitlementSetAuthorization(
 	memoryGauge common.MemoryGauge,
-	entitlementList []common.TypeID,
+	entitlementListConstructor func() []common.TypeID,
+	entitlementListSize int,
 	kind sema.EntitlementSetKind,
 ) EntitlementSetAuthorization {
 	common.UseMemory(memoryGauge, common.MemoryUsage{
 		Kind:   common.MemoryKindEntitlementSetStaticAccess,
-		Amount: uint64(len(entitlementList)),
+		Amount: uint64(entitlementListSize),
 	})
 
+	entitlementList := entitlementListConstructor()
```

Comment:
Maybe ensure the given count is an overestimate, and not and underestimate.

```suggestion
	entitlementList := entitlementListConstructor()
	if len(entitlementList) > entitlementListSize {
		panic()
	}
```